src/app/action-items/action-items.component.ts

```diff
@@ -16,36 +19,43 @@ export interface ActionItemMapped extends ActionItem {
 })
 export class ActionItemsComponent implements OnInit {
   dataSource: ActionItemMapped[];
-  isloadingActionItems = false;
-  private currentDate: Date = new Date();
+  isLoadingActionItems = false;
+  isLoadingActionItem = false;
+  currentDate: Date = new Date();
+  actionItemId: string;
+  actionItem: Observable<ActionItem>;
 
   constructor(
     private tasksService: TasksService,
     private daysCountService: DaysLeftToDeadlineService,
-    private matDialog: MatDialog
+    private matDialog: MatDialog,
+    private route: ActivatedRoute,
+    private router: Router,
+    private daysLeftPipe: DaysLeftCountedPipe
   ) {}
   ngOnInit() {
     this.retrieveActionItems();
+    this.route.queryParams.subscribe(params => {
```

Comment:
Please add unsubscribe handling. Also it would be nice to move it to the separate method.